src/types.rs

```diff
@@ -158,15 +157,15 @@ impl TypeDefinition {
             .filter(|c| matches!(c, Constraint::Occurs(_)))
             .next()
         {
-            return occurs.isl_occurs().try_into();
+            return Ok(occurs.occurs_range().to_owned());
         }
         // by default, if there is no `occurs` constraint for given type_def
         // then use `occurs:optional` if its `fields` constraint validation
         // otherwise, use `occurs: required` if its `ordered_elements` constraint validation
         if validation_constraint_name == "fields" {
-            return Range::optional();
+            return Ok(Range::optional());
         }
-        Range::required()
+        Ok(Range::required())
```

Comment:
Does this method still need to return `IonSchemaResult<Range>` or can it just return `Range`? I'm not seeing any `Err` propagation.

Reply:
Yes, definitely can be `Range`. I will change this.